src/geometry/ArborX_GeometryTraits.hpp

```diff
@@ -0,0 +1,101 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+#ifndef ARBORX_GEOMETRY_TRAITS_HPP
+#define ARBORX_GEOMETRY_TRAITS_HPP
+
+#include <Kokkos_Core.hpp>
+
+namespace ArborX
+{
+
+namespace GeometryTraits
+{
+
+struct PointTag
+{};
+
+struct BoxTag
+{};
+
+struct SphereTag
+{};
+
+template <typename Geometry>
+struct dimension
+{
+  using not_specialized = void; // tag to detect existence of a specialization
+};
+
+struct not_specialized
+{};
+
+template <typename Geometry>
+struct tag
+{
+  using type = not_specialized; // or void
```

Comment:
I'm surprised that you kept the comment.

Reply:
missed it